tests/integration/suite/scheduler/metrics/daprconnections.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/client"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd"
+	"github.com/dapr/dapr/tests/integration/framework/process/exec"
+	"github.com/dapr/dapr/tests/integration/framework/process/grpc/app"
+	"github.com/dapr/dapr/tests/integration/framework/process/logline"
+	"github.com/dapr/dapr/tests/integration/framework/process/scheduler"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(daprconnections))
+}
+
+type daprconnections struct {
+	scheduler *scheduler.Scheduler
+	daprd     *daprd.Daprd
+
+	connectionlogline *logline.LogLine
```

Comment:
Not a fan of using the log line here for testing as it is quite fragile. I think we can improve the test by running 3 daprds in `Run`, and doing a `Cleanup` of each. We should test that we observe the `dapr_scheduler_sidecars_connected_total` metric return `0, 1, 2, 3, 2, 1, 0`

Reply:
ok I wouldn't say its fragile, but I see what you are saying. I can work to make this better per your suggestion.

Can you re-send/share the comment you made about using a metric map. I vaguely remember seeing a comment from you about it, but am having a hard time finding it now.

Reply:
I noticed with the `Cleanup()` they werent shutting down in time, so what I did was test the metric by adding the 3 daprs and then just clean them up but not test the metric count going down.

Reply:
I got this working